tests/plugins/wysiwygarea/detached.js

```diff
@@ -0,0 +1,260 @@
+				var editorContainer = bot.editor.container,
+					editorContainerParent = editorContainer.getParent();
+
+				editorContainer.remove() ;
+				editorContainerParent.append( editorContainer );
+
+				CKEDITOR.tools.setTimeout( function() {
+					resume( function() {
+						var iframeElement = bot.editor.ui.space( 'contents' ).findOne( 'iframe' ),
+							editorData = iframeElement.getFrameDocument().getBody().getHtml();
+
+						assert.areEqual( '', editorData, 'Reattached editor. Editor data should be empty. Observing invalid DOM object.' );
+					} );
+				}, 500 );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor parent makes editor data empty with observed invalid DOM object': function() {
+			var startupData = '<p>CKEditor4</p>',
+				invalidObserveTarget = CKEDITOR.document.getById( 'invalidObserveTarget' ).$;
+
+			bender.editorBot.create( {
+				name: 'editor3',
+				startupData: startupData,
+				config: {
+					detachableParent: invalidObserveTarget
+				}
+			}, function( bot ) {
+				var editorContainerParent = bot.editor.container.getParent(),
+					parentParent = editorContainerParent.getParent();
+
+				editorContainerParent.remove() ;
+				parentParent.append( editorContainerParent );
+
+				CKEDITOR.tools.setTimeout( function() {
+					resume( function() {
+						var iframeElement = bot.editor.ui.space( 'contents' ).findOne( 'iframe' ),
+							editorData = iframeElement.getFrameDocument().getBody().getHtml();
+
+						assert.areEqual( '', editorData, 'Reattached editor parent. Editor data should be empty. Observing invalid DOM object.' );
+					} );
+				}, 500 );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor parent makes editor data empty with observed detached parent element': function() {
+			var startupData = '<p>CKEditor4</p>',
+				invalidObserveTarget = CKEDITOR.document.getById( 'editorDetachableParent2' );
+
+			bender.editorBot.create( {
+				name: 'editor4',
+				startupData: startupData,
+				config: {
+					detachableParent: invalidObserveTarget.$
+				}
+			}, function( bot ) {
+				invalidObserveTarget.remove() ;
+				CKEDITOR.document.getBody().append( invalidObserveTarget );
+
+				CKEDITOR.tools.setTimeout( function() {
+					resume( function() {
+						var iframeElement = bot.editor.ui.space( 'contents' ).findOne( 'iframe' ),
+							editorData = iframeElement.getFrameDocument().getBody().getHtml();
+
+						assert.areEqual( '', editorData, 'Reattached editor parent. Editor data should be empty. Observing detached element.' );
+					} );
+				}, 500 );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor with iframe content contains the same data with observed default dom object': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			var startupData = '<p>CKEditor4</p>';
+
+			bender.editorBot.create( {
+				startupData: startupData,
+				name: 'editor' + new Date().getTime(),
+				config: {
+					extraAllowedContent: 'iframe'
```

Comment:
I didn't notice where we use it? Shouldn't we provide custom startupData containing iframe in this test?

Reply:
Actually I bet you wanted to use fixture with `editor5` here as it contains `iframe` and is not used anywhere?